internal/providers/nscloud/api/rpc.go

```diff
@@ -292,45 +293,58 @@ func ListClusters(ctx context.Context, api API) (*KubernetesClusterList, error)
 
 func ExchangeToken(ctx context.Context, scopes ...string) (string, error) {
 	return tasks.Return(ctx, tasks.Action("nscloud.exchange-token"), func(ctx context.Context) (string, error) {
-		// Check if there is already tenant token stored.
-		tenantToken, err := auth.LoadTenantToken(ctx)
+		var username string
+		userAuth, err := auth.LoadUser()
 		if err == nil {
-			// If no scopes provided we can immediately return the token.
-			if len(scopes) == 0 {
-				return tenantToken.TenantToken, nil
-			}
+			username = userAuth.Username
+		} else if !(errors.Is(err, auth.ErrRelogin) && os.Getenv("GITHUB_ACTIONS") == "true") {
```

Comment:
Nit: `os.Getenv("GITHUB_ACTIONS") == "true"` should be a function in `internal/github`, similar to https://github.com/namespacelabs/foundation/blob/63ec8aa642dd072d658d8532a090cc6c08b2abcd/internal/environment/ci.go#L10-L11